pkg/server/updates.go

```diff
@@ -99,9 +105,31 @@ type versionInfo struct {
 
 // PeriodicallyCheckForUpdates starts a background worker that periodically
 // phones home to check for updates and report usage.
-func (s *Server) PeriodicallyCheckForUpdates() {
-	s.stopper.RunWorker(context.TODO(), func(ctx context.Context) {
-		startup := timeutil.Now()
+func (s *Server) PeriodicallyCheckForUpdates(ctx context.Context) {
+	startup := timeutil.Now()
+
+	builtins.Add("crdb_internal.force_diagnostic_report", []tree.Builtin{
```

Comment:
Oy no, please no. We're not ready to inject builtins this way.
Please add a static definition of the builtin in the builtins package, with either:
- a call to an exported function in this package, or,
- a call to a global function pointer defined in the `builtins` package, and initialized here

Reply:
We definitely can't export the func here and just have a builtin in `builtins` call it directly, based on dependency directions.

A mutable function pointer in `Builtins` that we update here *could* work, but that gets gross pretty quickly as we add more functions... and we already have a mutable map.

I think we're going to have to allow defining and registering builtins elsewhere -- a single package that containing all of them the will become will become a nightmare, both in terms of dependency management and the runtime state plumbing (or function pointer mutation).

Our single package with everything has worked so far, with our mostly pure functions plus a little plumbing in things like `evalCtx` for things like ClusterID or other system functions, but eventually (i.e. now?) think we want allow defining functions that maybe have more runtime dependencies where they can more cleanly capture those, rather than plumbing all the possible state and dependencies needed by any possible function to every function.

Reply:
Perhaps that is true, but it's too early to do so. In the coming release cycle we're going to change the builtin API and the evaluation strategy in major ways (in particular move away from the evaluation style where each stage of evaluation casts its arguments and allocates its own result values), and it is important to keep the cognitive overhead of doing so if that work could be guaranteed the builtin package is self-contained.

Reply:
What'd you think of putting it in the map literal with a dummy `Fn` and then swapping that function pointer v.s. introducing more package vars?

Reply:
I don't like it but I appreciate it can be a adeqate middleground. I do not have a better idea.

Reply:
Eh, I looked at it and the `Fn` field is hidden by the intermediate iface, so I guess i'm going the mutable func pointer route after all. Tried it with a little map to hold them, rather than individual top-level package vars though. RFAL.